### PR TITLE
add a Exception in hasAccess()

### PR DESCRIPTION
**Pull request by iamchenxin**
add a Exception when ($conf['remote']==false)  in function hasAccess() . 
To ensure the third rpc (like jsonrpc) get the right hint ( "RPC server not enabled" rather than "not authorized to call method'" )  ,when they using the remote.php and forgot check the $conf['remote'] themself.

### CODE REVIEW

**Comment by splitbrain** [comment]
Tests need to be updated to cope with the new behaviour. Can you please have a look at it?

**Comment by iamchenxin** [comment]
i'm sorry . maybe i have some misunderstanding with github . I do that jsonrpc in a different branch (name plugin not master). 
I did not realize that it will sync with pull requests . im so sorry .
My original goal was to amend a confusing return value to Exception in function hasAccess() in remote.php

Is that jsonrpc  useful ? I make it not only a jsonrpc server ,but also let the AJAX server  enable to process jsonrpc request . but i do not write the unit test for it yet . because im not farminar with php.(learn  php few months ago). If it is useful ,i should learn and try to write the unit test later.
i checkout to branch master .

**Comment by iamchenxin** [comment]
! oh, i am so sorry again . seems i carelessly commit informal code (jsonrpc) to master branch . 
sorry again . i should del it . and recommit again.
Hope i will not make things wrong this time.

File: lib/exe/xmlrpc.php [hunk 1]
```diff
@@ -35,7 +35,7 @@ function call($methodname, $args){
         } catch (RemoteAccessDeniedException $e) {
             if (!isset($_SERVER['REMOTE_USER'])) {
                 http_status(401);
-                return new IXR_Error(-32603, "server error. not authorized to call method $methodname");
+                return new IXR_Error(-32603, "server error. not authorized to call method $methodname|");
```
Comment (Klap-in) [inline lib/exe/xmlrpc.php]: What is the meaning of the vertical `|` after the method name?

**Comment by splitbrain** [comment]
I cherrypicked the appropriate commits. This also got rid of the weird ```|``` edit.

**Pull request closed without merging**